### PR TITLE
configurators: extend ConfiguratorBase to include more common fields …

### PR DESCRIPTION
**Pull request by Mic92**
…and dry up the code

## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/configurators/__init__.py [hunk 1]
```diff
@@ -34,15 +34,9 @@ def __init__(self):
...
-        if 'builders' not in c:
-            c['builders'] = []
-        self.builders = c['builders']
-        if 'workers' not in c:
-            c['workers'] = []
-        self.workers = c['workers']
+        fields = ['schedulers', 'protocols', 'builders', 'workers', 'projects', 'secretProviders']
+        for k in fields:
+            c.setdefault(k, [])
+            setattr(self, k, c[k])
```
[the remark this answers is not in the document]
Comment (Mic92) [reply master/buildbot/configurators/__init__.py]: done.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
@Mic92 The last changes broke tests. The result of setdefault does not need to be assigned.

**Comment by Mic92** [comment]
Yeah it should be self actually not c

**Comment by Mic92** [comment]
Now it's assigned to self instead to c

**Comment by mokibit** [comment]
Broken tests were fixed in PR https://github.com/buildbot/buildbot/pull/8031.
@p12tic  this PR can be closed.

**Comment by p12tic** [comment]
Thanks.

**Pull request closed without merging**